File: tarea1/generating_datasets.m
function generating_datasets(dataname)
%generating datasets receive the daname as input and return a set of files
%with the training and testing sets. The original.data file and the shuffle.m file should be either in
%the same directory of this function, or in a directory from the MATLAB
%path.
ns=20; %number of sets that we want to generate.
rng('default'); %we reset the random seed
if strcmp(dataname,'cereales') 
    orig_name='cereales.data';
elseif strcmp(dataname,'credit')
    orig_name='credit.data';
else disp('Error: data resquested is not in this database');
end
for i=1:ns
    A=load(orig_name); %load the original dataset
    [f,c]=size(A); %we obtain the number of rows and columns of A.
    B=shuffle(A);  %shuffle data
    %we choose a name for training and testing files
    name_tr=strcat(dataname,'-tr-',num2str(i),'.data'); 
    name_ts=strcat(dataname,'-ts-',num2str(i),'.data');
    %we write these files
    dlmwrite(name_tr,B([1:round(f*.75)],:),' ');
    dlmwrite(name_ts,B([(round(f*.75)+1):f],:),' ');
end

File: tarea2/generating_datasets.py
import numpy as np
import sys
import os

#dataname - string - dataset name
#ns - integer - number of sets to generate
#otype - string  - type of output sets (binary or text files)

def generating_datasets(dataname, ns=20, otype='binary'):
	if dataname == 'credit':
		orig_name = 'credit.data'
		#load original dataset as numpy array
		A = np.loadtxt(orig_name, dtype=float, delimiter=' ')
	elif dataname == 'diabetes':
		orig_name = 'diabetes.data'
		#load original dataset as numpy array
		A = np.loadtxt(orig_name, dtype=float, delimiter=',')
	else:
		 sys.error('Error: Requested data is not in this database')

	#rows and columns
	r,c = A.shape

	for i in range(ns):
		#shuffle rows of data matrix
		B = A.copy()
		np.random.shuffle(B)
		#names for training and testing files
		name_tr = dataname+'-tr-'+str(i)
		name_ts = dataname+'-ts-'+str(i)
		if otype=='binary':
			#save sets as numpy binary files
			np.save(name_tr, B[0:np.round(0.75*r)])
			np.save(name_ts, B[np.round(0.75*r):r])
		elif otype=='text':
			#save sets as text files
			np.savetxt(name_tr, B[0:np.round(0.75*r)], delimiter=' ')
			np.savetxt(name_ts, B[np.round(0.75*r):r], delimiter=' ')
	return 1


def generating_datasets2():
	path = './diabetes/'
	filenames = os.listdir(path)
	for filename in filenames:
		if 'ts' in filename: continue
		data = np.load(path+filename)
		#target name
		tgt = path+filename[:-4]+'{0}.npy'
		np.random.shuffle(data)
		np.save(tgt.format('-100'), data[0:100])
		np.random.shuffle(data)
		np.save(tgt.format('-200'), data[0:200])
		np.random.shuffle(data)
		np.save(tgt.format('-300'), data[0:300])
	return 1
